src/cpp/cpp_typecheck_expr.cpp

```diff
@@ -415,15 +414,18 @@ bool cpp_typecheckt::overloadable(const exprt &expr)
 
   forall_operands(it, expr)
   {
-    typet t=follow(it->type());
+    typet t = it->type();
 
     if(is_reference(t))
       t=t.subtype();
 
-    if(t.id()==ID_struct ||
-       t.id()==ID_union ||
-       t.id()==ID_c_enum || t.id() == ID_c_enum_tag)
+    if(
+      t.id() == ID_struct || t.id() == ID_union || t.id() == ID_c_enum ||
+      t.id() == ID_c_enum_tag || t.id() == ID_struct_tag ||
+      t.id() == ID_union_tag)
```

Comment:
This seems like a bug fix as well...

Reply:
Not one I am aware of, other than the `ID_c_enum_tag` having been a dead case before?